client/landing/stepper/hooks/use-is-domain-code-valid.ts

```diff
@@ -51,19 +53,26 @@ export function useIsDomainCodeValid( pair: DomainCodePair, queryOptions = {} )
 
 				// A `transferrability` property was added in D115244-code to check whether a mapped domain can be transferred
 				const isUnlocked =
-					[ 'transferrable', 'mapped_to_same_site_transferrable' ].includes(
+					[
+						domainAvailability.TRANSFERRABLE,
+						domainAvailability.TRANSFERRABLE_PREMIUM,
+						domainAvailability.MAPPED_SAME_SITE_TRANSFERRABLE,
+					].includes( availability.status ) ||
+					( [ domainAvailability.MAPPED, domainAvailability.MAPPED_OTHER_SITE_SAME_USER ].includes(
 						availability.status
-					) ||
-					( [ 'mapped_domain', 'mapped_to_other_site_same_user' ].includes( availability.status ) &&
-						'transferrable' === availability?.transferrability );
+					) &&
+						[ domainAvailability.TRANSFERRABLE, domainAvailability.TRANSFERRABLE_PREMIUM ].includes(
+							availability?.transferrability ?? ''
```

Comment:
```suggestion
							availability?.transferrability
```
I don't think the `?? ''` is necessary here as if `availability` is not defined, the `?.` operator will return `undefined` - and if `availability` is defined but `transferrability` is not, `undefined` will be returned too

Reply:
Well yeah but I get an error otherwise as `includes` only works with `string` and TypeScript throws an error